server/src/api/community.rs

```diff
@@ -472,7 +522,105 @@ impl Perform for Oper<EditCommunity> {
       res_sent.community.subscribed = None;
 
       ws.chatserver.do_send(SendCommunityRoomMessage {
-        op: UserOperation::EditCommunity,
+        op: UserOperation::DeleteCommunity,
+        response: res_sent,
+        community_id: data.edit_id,
+        my_id: ws.id,
+      });
+    }
+
+    Ok(res)
+  }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Perform for Oper<RemoveCommunity> {
+  type Response = CommunityResponse;
+
+  async fn perform(
+    &self,
+    pool: &DbPool,
+    websocket_info: Option<WebsocketInfo>,
+  ) -> Result<CommunityResponse, LemmyError> {
+    let data: &RemoveCommunity = &self.data;
+
+    let claims = match Claims::decode(&data.auth) {
+      Ok(claims) => claims.claims,
+      Err(_e) => return Err(APIError::err("not_logged_in").into()),
+    };
+
+    let user_id = claims.id;
+
+    // Check for a site ban
+    let user = blocking(pool, move |conn| User_::read(conn, user_id)).await??;
+    if user.banned {
+      return Err(APIError::err("site_ban").into());
+    }
```

Comment:
Is this needed here? I would just make sure that the admin rank is removed if an admin is banned (assuming that is even possible).

Reply:
tbh I'm not sure, but its probably best to be safe... we'd need to open an issue because I havent tested what happens when you ban a lesser admin, and their user row has `admin: true`, `banned: true`